
01/26/2024 15:12:42 - WARNING - __main__ - Process rank: 0, device: cuda:0, n_gpu: 1distributed training: True, 16-bits training: False
/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/load.py:2483: FutureWarning: 'use_auth_token' was deprecated in favor of 'token' in version 2.14.0 and will be removed in 3.0.0.
You can remove this warning by passing 'token=<use_auth_token>' instead.
  warnings.warn(
[INFO|configuration_utils.py:667] 2024-01-26 15:12:44,915 >> loading configuration file /home/ubuntu/Documents/ai/model/ChatGLM3/chatglm3-6b/config.json
[INFO|configuration_utils.py:667] 2024-01-26 15:12:44,920 >> loading configuration file /home/ubuntu/Documents/ai/model/ChatGLM3/chatglm3-6b/config.json
[INFO|configuration_utils.py:725] 2024-01-26 15:12:44,921 >> Model config ChatGLMConfig {
  "_name_or_path": "/home/ubuntu/Documents/ai/model/ChatGLM3/chatglm3-6b",
  "add_bias_linear": false,
  "add_qkv_bias": true,
  "apply_query_key_layer_scaling": true,
  "apply_residual_connection_post_layernorm": false,
  "architectures": [
    "ChatGLMModel"
  ],
  "attention_dropout": 0.0,
  "attention_softmax_in_fp32": true,
  "auto_map": {
    "AutoConfig": "configuration_chatglm.ChatGLMConfig",
    "AutoModel": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForCausalLM": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForSeq2SeqLM": "modeling_chatglm.ChatGLMForConditionalGeneration",
    "AutoModelForSequenceClassification": "modeling_chatglm.ChatGLMForSequenceClassification"
  },
  "bias_dropout_fusion": true,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "ffn_hidden_size": 13696,
  "fp32_residual_connection": false,
  "hidden_dropout": 0.0,
  "hidden_size": 4096,
  "kv_channels": 128,
  "layernorm_epsilon": 1e-05,
  "model_type": "chatglm",
  "multi_query_attention": true,
  "multi_query_group_num": 2,
  "num_attention_heads": 32,
  "num_layers": 28,
  "original_rope": true,
  "pad_token_id": 0,
  "padded_vocab_size": 65024,
  "post_layer_norm": true,
  "pre_seq_len": null,
  "prefix_projection": false,
  "quantization_bit": 0,
  "rmsnorm": true,
  "seq_length": 8192,
  "tie_word_embeddings": false,
  "torch_dtype": "float16",
  "transformers_version": "4.30.2",
  "use_cache": true,
  "vocab_size": 65024
}
[INFO|tokenization_utils_base.py:1821] 2024-01-26 15:12:44,924 >> loading file tokenizer.model
[INFO|tokenization_utils_base.py:1821] 2024-01-26 15:12:44,924 >> loading file added_tokens.json
[INFO|tokenization_utils_base.py:1821] 2024-01-26 15:12:44,925 >> loading file special_tokens_map.json
[INFO|tokenization_utils_base.py:1821] 2024-01-26 15:12:44,925 >> loading file tokenizer_config.json
[INFO|modeling_utils.py:2575] 2024-01-26 15:12:44,970 >> loading weights file /home/ubuntu/Documents/ai/model/ChatGLM3/chatglm3-6b/pytorch_model.bin.index.json
[INFO|configuration_utils.py:577] 2024-01-26 15:12:44,970 >> Generate config GenerationConfig {
  "_from_model_config": true,
  "eos_token_id": 2,
  "pad_token_id": 0,
  "transformers_version": "4.30.2"
}



Loading checkpoint shards:  86%|██████████████████████████████████████████████████████████▎         | 6/7 [00:06<00:01,  1.05s/it]
Loading checkpoint shards: 100%|████████████████████████████████████████████████████████████████████| 7/7 [00:06<00:00,  1.01it/s]
[INFO|modeling_utils.py:3295] 2024-01-26 15:12:51,929 >> All model checkpoint weights were used when initializing ChatGLMForConditionalGeneration.
[WARNING|modeling_utils.py:3297] 2024-01-26 15:12:51,929 >> Some weights of ChatGLMForConditionalGeneration were not initialized from the model checkpoint at /home/ubuntu/Documents/ai/model/ChatGLM3/chatglm3-6b and are newly initialized: ['transformer.prefix_encoder.embedding.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
[INFO|modeling_utils.py:2927] 2024-01-26 15:12:51,930 >> Generation config file not found, using a generation config created from the model config.
prefix:
Running tokenizer on prediction dataset:   0%|                                                      | 0/10 [00:00<?, ? examples/s]
Traceback (most recent call last):
  File "/home/ubuntu/Documents/ai/ChatGLM2-6B/ptuning/main.py", line 480, in <module>
    main()
  File "/home/ubuntu/Documents/ai/ChatGLM2-6B/ptuning/main.py", line 294, in main
    predict_dataset = predict_dataset.map(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 592, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 557, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3093, in map
    for rank, done, content in Dataset._map_single(**dataset_kwargs):
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3470, in _map_single
    batch = apply_function_on_filtered_inputs(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3349, in apply_function_on_filtered_inputs
    processed_inputs = function(*fn_args, *additional_args, **fn_kwargs)
  File "/home/ubuntu/Documents/ai/ChatGLM2-6B/ptuning/main.py", line 187, in preprocess_function_eval
    prompt = tokenizer.build_prompt(query, history)
AttributeError: 'ChatGLMTokenizer' object has no attribute 'build_prompt'
Traceback (most recent call last):
  File "/home/ubuntu/Documents/ai/ChatGLM2-6B/ptuning/main.py", line 480, in <module>
    main()
  File "/home/ubuntu/Documents/ai/ChatGLM2-6B/ptuning/main.py", line 294, in main
    predict_dataset = predict_dataset.map(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 592, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 557, in wrapper
    out: Union["Dataset", "DatasetDict"] = func(self, *args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3093, in map
    for rank, done, content in Dataset._map_single(**dataset_kwargs):
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3470, in _map_single
    batch = apply_function_on_filtered_inputs(
  File "/home/ubuntu/miniconda3/envs/chatglm2/lib/python3.10/site-packages/datasets/arrow_dataset.py", line 3349, in apply_function_on_filtered_inputs
    processed_inputs = function(*fn_args, *additional_args, **fn_kwargs)
  File "/home/ubuntu/Documents/ai/ChatGLM2-6B/ptuning/main.py", line 187, in preprocess_function_eval
    prompt = tokenizer.build_prompt(query, history)
AttributeError: 'ChatGLMTokenizer' object has no attribute 'build_prompt'